src/lib/diagnostics.ts

```diff
@@ -89,6 +89,23 @@ export class DiagnosticManager {
             target: vscode.Uri.parse(violation.url),
             value: violation.ruleName
         } : violation.ruleName;
+        if (engine === 'apexguru') {
+            const apexGuruViolation = violation as ApexGuruViolation;
+        
+            if (apexGuruViolation.suggestedCode?.trim()) {
```

Comment:
If the suggestedCode is empty... maybe that is because we are suggesting to replace it. Can we instead just explicitly check if the suggestedCode is undefined as a way of determining this.

Reply:
Ah! this is a super-edge case. But valid point. Let me take care of this.

Reply:
Done!